browser/src/service/NodeAutosuggestService.tsx

```diff
@@ -1,15 +1,69 @@
 import React from "react"
 
-export class NodeAutosuggestService extends React.Component{
-    
-    add(a: number, b: number){
-        return a+b
+export class NodeAutosuggestService {
+    //constructor is empty, only necessary to access methods parseString, and addQuestionsMarks
+    constructor() {}
+
+    /* 
+    Component is passed in and a shorter name of the component is returned, 
+    Time Complexity O(n)
+    Space Complexity O(n)
+    */
+    parseString(component: string) {
+        var shorter_component = ""
+        for (var i = 0; i < component.length; i++) {
```

Comment:
this is pretty hard to follow-- could it be more token-based,  eg given your current processing state, are you looking for a qualifier (start of input, starts with `@`,  a general type name,  a type parameter/generic (`<>`), type parameter bounds (`?` inside of type parameter)

Reply:
Agree, the logic here is more like ACM-style, considering making it more readable and it's easier for others to maintain in the future.